bindings/rust/s2n-tls/src/testing/s2n_tls.rs

```diff
@@ -892,4 +892,55 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn key_updates() -> Result<(), Error> {
+        let config = {
+            let config = config_builder(&security::DEFAULT_TLS13)?;
+            config.build()?
+        };
+
+        let server = {
+            let mut server = crate::connection::Connection::new_server();
+            server.set_config(config.clone())?;
+            Harness::new(server)
+        };
+
+        let client = {
+            let mut client = crate::connection::Connection::new_client();
+            client.set_config(config)?;
+            Harness::new(client)
+        };
+
+        let pair = Pair::new(server, client);
+        let mut pair = poll_tls_pair(pair);
+
+        // there haven't been any key updates at the start of the connection
+        #[cfg(feature = "unstable-ktls")]
+        {
+            let client_updates = pair.client.0.connection.as_ref().key_update_counts()?;
+            assert_eq!(client_updates, (0, 0));
+            let server_updates = pair.server.0.connection.as_ref().key_update_counts()?;
+            assert_eq!(server_updates, (0, 0));
+        }
+
+        pair.server
+            .0
+            .connection
+            .as_mut()
+            .request_key_update(PeerKeyUpdate::KeyUpdateNotRequested)?;
+        assert!(pair.poll_send(Mode::Server, &[0]).is_ready());
+
+        // the server send key has been updated
+        #[cfg(feature = "unstable-ktls")]
```

Comment:
Why not gate the entire test on unstable-ktls? It's not exactly useful otherwise